### DISCUSSION TITLE
ENSEMBLE_STAT_ENSEMBLE_FLAG_MEAN entry

### DISCUSSION THREAD

**Discussion by michelleharrold** [dtcenter/METplus#1560] [Configuration] [answered]
Good Morning!

I am running ensemble-stat via METplus, and I do not want any NetCDF ensemble product output. This is what I have set in my configuration file:

```
ENSEMBLE_STAT_ENSEMBLE_FLAG_LATLON = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_MEAN = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_STDEV = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_MINUS = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_PLUS = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_MIN = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_MAX = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_RANGE = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_VLD_COUNT = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_FREQUENCY = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_NEP = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_NMEP = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_RANK = FALSE
ENSEMBLE_STAT_ENSEMBLE_FLAG_WEIGHT = FALSE
```

However, when I run, I am getting ENS_MEAN output in NetCDF files for all variable/level combos (no other output types, just mean). Is there a way to turn off the product generation in ensemble-stat METplus configuration file? I am using this version of METplus on Cheyenne: /glade/u/home/mccabe/METplus.bugfix_1555

Thanks!
Michelle Harrold

**Comment by jprestop** [comment, marked as answer]
Hi @michelleharrold.  I consulted with @JohnHalleyGotway about it, and he had some very useful information. 

If you inspects the log output from Ensemble-Stat, she should find this warning message:
```
WARNING: 
WARNING: process_command_line() -> enabling NetCDF ensemble mean computation to be used in verification.
WARNING: 
```
Since you are not providing an external ensemble mean file on the command line using the "-ens_mean" option, the Ensemble-Stat must calculate that ensemble mean field on the fly from the input data. The way it currently does that is by writing that mean field out to the NetCDF file and then reading it back in later when it's needed. If ensemble_flag.mean = FALSE in the user config file, but the ensemble mean is needed by the tool, it resets that flag to true and prints that warning message.

So it's behaving as it was designed to behave. That being said, for MET version 11.0.0, [dtcenter/MET#1908](https://github.com/dtcenter/MET/issues/1908) will remove the ensemble post-processing from the Enesmble-Stat tool. That would be a good opportunity to revise the logic for handling the ensemble-mean field, if needed. You could add a comment to that issue to make note of this request so we could consider it during development.

I hope that helps.  Please let us know if you have any additional questions.

**Comment by michelleharrold** [comment]
@jprestop - thanks for the thoughtful and thorough response! I was unaware that MET wrote out a mean file (I assumed it did the calculations internally). I have run gen-ens-prod in a previous step, so I have the ensemble mean data from that step that it sounds like I can pass in.

Thanks!

**Reply by jprestop** [reply]
Thanks, @michelleharrold.  At this point I'm going to LOCK this conversation to prevent future posts. Our team has decided that once discussions have been answered, and the answer has been confirmed, we'll lock them. We want to encourage users to ask new questions in new discussions rather than posting to old ones. Hopefully that'll make the questions/answers easier for other users to follow. So if/when more issues/questions arise, please feel free to start a new discussion.
